vizro-core/tests/unit/vizro/models/_components/form/test_checklist.py

```diff
@@ -0,0 +1,144 @@
+        assert checklist.options == ["A", "B", "C"]
+        assert checklist.value == ["A"]
+        assert checklist.title == "Title"
+        assert checklist.actions == []
+
+    @pytest.mark.parametrize(
+        "test_options, expected",
+        [
+            (["A", "B", "C"], ["A", "B", "C"]),
+            ([1, 2, 3], [1.0, 2.0, 3.0]),
+            ([1.0, 2.0, 3.0], [1.0, 2.0, 3.0]),
+            ([True, False, True], [True, False, True]),
+            (
+                [{"label": "A", "value": "A"}, {"label": "B", "value": "B"}],
+                [{"label": "A", "value": "A"}, {"label": "B", "value": "B"}],
+            ),
+            (
+                [{"label": "New York", "value": "NYC"}, {"label": "Berlin", "value": "BER"}],
+                [{"label": "New York", "value": "NYC"}, {"label": "Berlin", "value": "BER"}],
+            ),
+            (
+                [{"label": "True", "value": True}, {"label": "False", "value": False}],
+                [{"label": "True", "value": True}, {"label": "False", "value": False}],
+            ),
+            ([True, 2.0, 1.0, "A", "B"], ["True", "2.0", "1.0", "A", "B"]),
+        ],
+    )
+    def test_create_checklist_valid_options(self, test_options, expected):
+        checklist = Checklist(options=test_options)
+
+        assert hasattr(checklist, "id")
+        assert checklist.type == "checklist"
+        assert checklist.options == expected
+        assert checklist.value is None
+        assert checklist.title is None
+        assert checklist.actions == []
+
+    @pytest.mark.parametrize("options", [1, "A", True, 1.0])
+    def test_create_checklist_invalid_options_type(self, options):
+        with pytest.raises(ValidationError, match="value is not a valid list"):
+            Checklist(options=options)
+
+    def test_create_checklist_invalid_options_dict(self):
+        with pytest.raises(
+            ValidationError,
+            match="Invalid argument `options` passed into Selector. Expected a dict with keys `label` and `value`.",
+        ):
+            Checklist(options=[{"hello": "A", "world": "A"}, {"hello": "B", "world": "B"}])
+
+    @pytest.mark.parametrize(
+        "options, test_value",
+        [
+            (["A", "B", "C"], ["A"]),
+            ([1, 2, 3], [1, 2]),
+            ([1.0, 2.0, 3.0], [1.0, 2.0]),
+            ([True, False], [False, True]),
+            ([{"label": "A", "value": "A"}, {"label": "B", "value": "B"}], ["A", "B"]),
+            ([True, 2.0, 1.0, "A", "B"], ["True", "A"]),
+        ],
+    )
+    def test_create_checklist_valid_value(self, options, test_value):
+        checklist = Checklist(options=options, value=test_value)
+
+        assert hasattr(checklist, "id")
+        assert checklist.type == "checklist"
+        assert checklist.value == test_value
+        assert checklist.title is None
+        assert checklist.actions == []
+
+    @pytest.mark.parametrize(
+        "test_value, options",
```

Comment:
Can we preserve parameter order in those two tests (or

"options, test_value", or
"test_value, options",
?